### DISCUSSION TITLE
Where to find cable information (S21)

### DISCUSSION THREAD

**Discussion by pito111** [revk/xESP32-Faikout#81] [Q&A] [answered]
Hello,

amazon product page references webpage (faikin.revk.uk) which redirects to this repo regarding cable information but I am unable to find it. I received the module and it is without cable and I cannot find the information in this repository. Maybe I am "blind" but could you please direct me where to find it?

Thank you.

Best,

Stefan

**Comment by revk** [comment, marked as answer]
The listing does say "No housing or leads supplied". You need a S21 cable. It depends on your make of air-con.

Daikin sell cables I believe, but again, depends on your model. People sell them, e.g. [Ebay](https://www.ebay.co.uk/itm/115800273576)

Some people have used "header wires", like [Amazon](https://www.amazon.co.uk/Jumper-Multicolored-Breadboard-Compatible-Arduino/dp/B0C4H93T2T)

The pin out is on [List of conformed working air-con](https://github.com/revk/ESP32-Faikin/wiki/List-of-confirmed-working-air-con-units)

**Reply by pito111** [reply]
Thank you.

The "header wires" are a good idea for testing at least (I have some).

I was trying to find information about the connector type in order to maybe make the cable myself. I see in the kicad files JST EH series connector, S5B-EH (http://www.jst-mfg.com/product/pdf/eng/eEH.pdf). Is this correct? Do you know if the other side (S21 Daikin) is the same? Sorry for maybe a stupid question but I burned myself with simillar connectors in the past when I did not know the exact model and bought ones that slightly did not fit because there are aparently multiple similar ones but slightly different.

BTW: I was not complaining that the cable is not there (I was aware of that). Sorry for clumsy writing.

**Reply by revk** [reply]
I think the plug/socket is the same both ends - check your PCB for an connector labeled S21.

Yes, it is the JST EH - this is the one used on the board: [JST EH](https://datasheet.lcsc.com/lcsc/1810261108_JST-Sales-America-S5B-EH-LF-SN_C265103.pdf)

Thanks.

**Comment by pito111** [comment]
Just an update if somebody looks here with similar situation

New Daikin AC split units (Models I have are FTXM25R and CTXM15R - Daikin Perfera 2022) have a different S21 port https://github.com/revk/ESP32-Faikin/discussions/49 P/N: PAP-05V-S. Daikin sells an adapter cable. Product code [EKRS21](https://www.daikin.eu/en_us/products/product.html/EKRS21.html).

With that, I was able to make it work. I had to connect all 5 pins. Even the suggested Daikin cable has only 4 wires, I was not able to make Faikin to work with 4. I tried, I think, all combinations but the best I was able to get was that it booted up but reported `not connected` when trying to control the unit.

The second thing is, that I had a Daikin wifi module there and was able to replace it with Faikin after cutting the PCB to size. I suspect the place to be there even if you do not have a Wifi module included. Dimensions are 45mm X 36mm.

Link to pictures for better understanding.
https://mega.nz/folder/aBlETAwZ#UwIzLawwljJ0Xh9-jTLTZw

**Reply by revk** [reply]
Thanks. Just to be clear, the Faikin board is intended to be snapped out of that larger board, and is tiny. But that trick does mean you can just mount it?, which is neat.

**Reply by pito111** [reply]
Exactly. I snapped only the part on connector side, the resulting size is close enough to fit the long horizontal dimension. Then I cut from sides to fit vertical size and I was able to put it directly as a replacement to the original Daikin module. It fits perfectly. Route a cable to S21 connector and you are golden.

**Reply by revk** [reply]
Nice

**Reply by revk** [reply]
I wonder if I should add extra V-cuts so you can just snap off some of the bits - do you have the exact dimensions of the original board, and I'll see what I can do to the design.

Edit: You said Dimensions are 45mm X 36mm.

**Reply by pito111** [reply]
It could be nice to have the cuts (a little bit of time an mess saving) if the dimensions are universal enough (I would guess they would be) and barring Daikin changes them too often.

Anyway, it is not a big deal as it is to cut it IMO.

**Reply by pito111** [reply]
I just got to look at the new PCBs I got a few weeks earlier. It seems now I will have to cut all sides since you changed the dimensions and removed the "vertical" V-cuts. Anyway, rotary tool should do the trick, but maybe those additional V-cuts would be nice since now I have to do twice the work :D.

It will be late for me since I have enough of them, but for the "next guy" in future batches it could be nice (keeping in mind the points I made earlier) to include those additional cuts for 45mmX36mm. Maybe a research/poll if this is the only dimension that is present on newer units or if it is all over the place for some reason could be helpful.

**Reply by revk** [reply]
I think we have a design now. JLC we're being a problem with vcuts. So future boards should be a bit easier.

![image](https://github.com/revk/ESP32-Faikin/assets/996983/2e970fe3-4163-4a71-afdf-8f8c388ae023)

**Reply by pito111** [reply]
I would love to have that right now o:-). I am "playing" with razor knife and file to make it the correct size (less mess then rotary tool).

Anyway, it looks great.

**Comment by njh** [comment]
There have been a number of similar enquiries about buying / making cables.

I had my own thread with pictures here:
https://github.com/revk/ESP32-Faikin/discussions/49
But it is quite hard to find, with discussions closed, when the original author is sorted out.

I wonder if a Wiki page with details of where to buy a cable, or the parts needed to make a cable, would be a good idea.

**Reply by revk** [reply]
That sounds like a good idea - the wiki is open for anyone to edit.

**Comment by BenJamesAndo** [comment]
Would this S21 cable work with the Faikin?
https://www.saturnsales.co.uk/Daikin-EKRS21-Wiring-Harness.html
It seems to be the cheapest place I can find them, being £10.00.
However I'm based in Australia and they don't ship here. To get an S21 cable would cost me about £43.99.

This website also sells it, but they call it the new S21 port. My aircon is the FTXS35GVMA from 2009 so I doubt it would be using a new S21 port.
https://aircon-online.co.uk/product/wire-harness-for-new-s21-port/

**Reply by revk** [reply]
First cable is male to female by the look of it, an extension. So no, that won't work.

Second link is not showing me any images.

I think I suggested jumper wires as a simple (cheap) alternative.

**Reply by pito111** [reply]
The first one is adapter cable from new style S21 ports, like I have to the old style like Faikin uses (and older Daikin models). I bought some, but finally decided to create my own cables (thanks to another thread that explains which connectors and stuff are needed).

The second one the same as revk (no picture) so no idea.

**Reply by BenJamesAndo** [reply]
Thank you both for the info.
@pito111 could you link to the thread that explains how to make the connectors yourself? I have never made my own cables before but if it explains everything in detail then I might be able to do it.

**Reply by pito111** [reply]
This one mentioned above should be it https://github.com/revk/ESP32-Faikin/discussions/49

It was also first time for me making this kind of cable, but manageable. Order few extra just in case ;)

**Comment by RagingRoosevelt** [comment]
I had trouble finding the JST EH parts needed on amazon.  On DigiKey, they are [#455-1003-ND](https://www.digikey.com/en/products/detail/jst-sales-america-inc/EHR-5/527227) (JST EHR-5) and [#455-1042-1-ND](https://www.digikey.com/en/products/detail/jst-sales-america-inc/SEH-001T-P0-6/527267) (the crimp socket).  I also used some cuts off my [22gauge silicone 6p flex cable](https://www.amazon.com/gp/product/B09R1KDFHZ/) spool and just stripped down to 5 connectors.  

A word of caution: make sure if one of your leads is in the ground position on one plug when plugged into the faikin, if you plug the other end into the fiakin, that it's still the ground wire.  For me, that meant that I had to wire one side "backwards" for it to work right.  Just make sure to double check before plugging in.
